### DISCUSSION TITLE
How to link to co-authors consistently throughout?

### DISCUSSION THREAD

**Discussion by catanzaromj** [alshedivat/al-folio#2213] [Q&A] [answered]
I have a couple questions regarding co-authors and linking to them. 

First, what is the correct capitalization for the `_data/coauthors.yml` file? The sample file in`_data/coauthors.yml` has lower case last names and upper case first names. This capitalization is the only one I can get to (partially) work. In `CUSTOMIZE.md`, the author last names are capitalized. When I try that, none of the authors get linked to.

Second, in the case when I do use some capitalization and some links appear, the same author *sometimes* gets a link and *sometimes* does not, even though their name is seemingly spelled exactly the same with the same punctuation and spacing. Does the author order matter? Are authors not linked if they are "hidden" behind "and Z more authors"?
For example, in this screen shot, see "Vladimir Y. Chernyak" 
![image](https://github.com/alshedivat/al-folio/assets/25447423/2aecd7f6-f937-4284-a2cf-537dd13c462a)

The first author bib entry is:
author = {Li, Hao and Catanzaro, Michael J. and Tretiak, Sergei and Chernyak, Vladimir Y.},

The second author bib entry is:
author = {Catanzaro, Michael J. and Chernyak, Vladimir Y. and Klein, John R.},

Thanks for any help or clarification!

**Comment by george-gca** [comment, marked as answer]
> In CUSTOMIZE.md, the author last names are capitalized. When I try that, none of the authors get linked to.

This was wrong. Just fixed that, thanks for noticing.

The answer is in [_layouts/bib.liquid](https://github.com/alshedivat/al-folio/blob/e430a8949c1fe687ded81fac7786c2872f98f290/_layouts/bib.liquid#L63). More specifically:

```liquid
      {%- assign clean_last_name = author_last_name | downcase | remove_accents -%}
        {% if site.data.coauthors[clean_last_name] %}
          {%- for coauthor in site.data.coauthors[clean_last_name] -%}
            {% if coauthor.firstname contains author.first %}
              {%- assign coauthor_url = coauthor.url -%}
```

We normalize the author last name and search for an entry in [_data/coauthors.yml](https://github.com/alshedivat/al-folio/blob/master/_data/coauthors.yml), so the last names here must be lower case and without accents.

If it finds, it iterates through all possible first names and checks if one applies.

> Does the author order matter? Are authors not linked if they are "hidden" behind "and Z more authors"?

The order doesn't matter, but authors hidden in "more authors" will not link to their urls. You can check in [_layouts/bib.liquid](https://github.com/alshedivat/al-folio/blob/e430a8949c1fe687ded81fac7786c2872f98f290/_layouts/bib.liquid#L96) that we don't use `‎coauthor_url‎` variable when doing that part.

**Reply by catanzaromj** [reply]
Gotcha, thanks.

**Reply by catanzaromj** [reply]
@george-gca My impression is that I could mimic the logic you have in the code block above to get the author url to show for the hidden authors. Do you see any issues with this? Is there a reason it wasn't implemented this way from the beginning?

**Reply by george-gca** [reply]
I wasn't here from the beginning, so no idea. But the current implementation uses the whole name of the other authors as a clickable area to hide the names again. You could or add the hide option to another place or make it once shown always shown. But the easiest way I think is just increase the number of authors that it displays by default. For me that was enough.

**Reply by catanzaromj** [reply]
Oh great suggestion. Thanks again!
